src/CMakeLists.txt

```
@@ -39,54 +41,113 @@ set(rime_core_deps
     ${Glog_LIBRARY}
     ${YamlCpp_LIBRARY}
     ${CMAKE_THREAD_LIBS_INIT}
-    ${rime_optional_deps}
-    ${rime_plugins_deps})
-set(rime_extra_deps
-    ${ICONV_LIBRARIES}
+    ${rime_optional_deps})
+set(rime_dict_deps
     ${LevelDb_LIBRARY}
-    ${Marisa_LIBRARY}
+    ${Marisa_LIBRARY})
+set(rime_gears_deps
+    ${ICONV_LIBRARIES}
     ${Opencc_LIBRARY})
+set(rime_levers_deps "")
 
 if(MINGW)
   set(rime_core_deps ${rime_core_deps} wsock32 ws2_32)
 endif()
 
 if(BUILD_SEPARATE_LIBS)
   set(rime_deps ${rime_core_deps})
-  set(rime_gears_deps ${rime_library} ${rime_extra_deps})
 else()
-  set(rime_deps ${rime_core_deps} ${rime_extra_deps})
+  set(rime_deps
+    ${rime_core_deps}
+    ${rime_dict_deps}
+    ${rime_gears_deps}
+    ${rime_levers_deps}
+    ${rime_plugins_deps})
 endif()
 
 if(BUILD_SHARED_LIBS)
-  add_library(rime ${rime_plugins_objs} ${rime_src})
+  add_library(rime ${rime_src})
   target_link_libraries(rime ${rime_deps})
-  set_target_properties(rime PROPERTIES DEFINE_SYMBOL "RIME_EXPORTS")
-  set_target_properties(rime PROPERTIES VERSION ${rime_version} SOVERSION ${rime_soversion})
+  set_target_properties(rime PROPERTIES
+    DEFINE_SYMBOL "RIME_EXPORTS"
+    VERSION ${rime_version}
+    SOVERSION ${rime_soversion})
   if(XCODE_VERSION)
     set_target_properties(rime PROPERTIES INSTALL_NAME_DIR "@rpath")
   endif()
   install(TARGETS rime DESTINATION ${LIB_INSTALL_DIR})
+
   if(BUILD_SEPARATE_LIBS)
-    add_library(rime-gears ${librime_gears_src})
-    target_link_libraries(rime-gears ${rime_gears_deps})
-    add_dependencies(rime-gears ${rime_library})
-    set_target_properties(rime-gears PROPERTIES VERSION ${rime_version} SOVERSION ${rime_soversion})
+    add_library(rime-dict ${rime_dict_module_src})
+    target_link_libraries(rime-dict
+      ${rime_dict_deps}
+      ${rime_library})
+    add_dependencies(rime-dict ${rime_library})
+    set_target_properties(rime-dict PROPERTIES
+      VERSION ${rime_version}
+      SOVERSION ${rime_soversion})
+    if(XCODE_VERSION)
+      set_target_properties(rime-dict PROPERTIES INSTALL_NAME_DIR "@rpath")
+    endif()
+    install(TARGETS rime-dict DESTINATION ${LIB_INSTALL_DIR})
+
+    add_library(rime-gears ${rime_gears_src})
+    target_link_libraries(rime-gears
```

```suggestion
    target_link_libraries(rime-gears ${rime_gears_deps})
```
There is no need to link `${rime_library}`, `${rime_gears_deps}`, `${rime_dict_library}`.

I have tested it in Linux, and it should work in macOS.

---

No this doesn't work for the macOS build.
Errors were cannot find symbol in rime libraries it depends on.

---

Maybe the solution is 
```
target_link_libraries(rime-gears ${rime_gears_deps})
IF(APPLE)
  SET_TARGET_PROPERTIES(rime-gears PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
ENDIF(APPLE)
```

References:
[1] https://stackoverflow.com/questions/42722759/portable-plugins-with-cmake
[2] https://stackoverflow.com/questions/36662920/xcode-clang-link-build-dynamic-framework-or-dylib-not-embed-dependencies